misc/python/materialize/scalability/workloads.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from materialize.scalability.operation import Operation
+from materialize.scalability.operations import (
+    InsertDefaultValues,
+    SelectCount,
+    SelectOne,
+    SelectStar,
+    SelectUnionAll,
+)
+from materialize.scalability.workload import Workload
+
+
+class InsertWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [InsertDefaultValues()]
+
+
+class SelectOneWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [SelectOne()]
+
+
+class SelectStarWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [SelectStar()]
+
+
+class SelectCountWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [SelectCount()]
+
+
+class SelectUnionAllWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [SelectUnionAll()]
+
+
+class InsertAndSelectCountWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [InsertDefaultValues(), SelectCount()]
+
+
+class InsertAndSelectStarWorkload(Workload):
+    def operations(self) -> list["Operation"]:
+        return [InsertDefaultValues(), SelectStar()]
```

Comment:
Is it intended that the `SelectStar()` operations gets more expensive with each insert?

Reply:
Yes, this will indeed happen. I have not played much with this scenario, I will see how useful it is in practice. Maybe `SELECT COUNT(*)` or `SELECT * LIMIT 1` or something along those lines will allow one to run selects mixed with inserts while isolating the individual select from the size of the dataset. Either way, we are talking about 2K rows inserted here, so if a SELECT takes a lot of time to run, it would be an issue given how few rows will be there.